src/data/markdown/docs/02 javascript api/02 k6/group- name- fn -.md

```diff
@@ -17,6 +17,19 @@ Run code inside a group. Groups are used to organize results in a test.
 | ---- | ------------------------- |
 | any  | The return value of _fn_. |
 
+<Blockquote mod="warning">
+
+Avoid using `group` with async functions or asynchronous code.
+If you do, k6 might apply tags in a way that is unreliable or unintuitive.
+
+If you start promise chains or even use `await` within `group`, some code within the group will be waited for and tagged with the proper `group` tag, but others won't be.
```

Comment:
```suggestion
If you start promise chains or even use `await` within `group`, some emitted metrics within the group will be waited for and tagged with the proper `group` tag, but others won't be.
```

Reply:
the problmes are two:
1. code won't we waited for or measured 
2. the metrics emitted by that code might be tagged "correctly" or not 

In practice we either need to split those apart and explain both of the in this snippet or I prefer to keep it as is. And given that explaining the whole problem is more adequote for the issue I prefer to do it there.

Reply:
fair enough, and I misread something when making the suggestion :sweat_smile:  :man_facepalming: